pkgs/applications/science/logic/coq/default.nix

```diff
@@ -43,13 +43,13 @@ let
    "8.12.2".sha256     = "18gscfm039pqhq4msq01nraig5dm9ab98bjca94zldf8jvdv0x2n";
    "8.13.0".sha256     = "0sjbqmz6qcvnz0hv87xha80qbhvmmyd675wyc5z4rgr34j2l1ymd";
    "8.13.1".sha256     = "0xx2ns84mlip9bg2mkahy3pmc5zfcgrjxsviq9yijbzy1r95wf0n";
-   "8.13.2".sha256     = "1884vbmwmqwn9ngibax6dhnqh4cc02l0s2ajc6jb1xgr0i60whjk";
+   "8.13.2".sha256     = "14d4alp35hngvga9m7cfp5d1nl62xdj0nm4811f2jjblk86gxxk4";
    "8.14.0".sha256     = "04y2z0qyvag66zanfyc3f9agvmzbn4lsr0p1l7ck6yjhqx7vbm17";
-   "8.14.1".sha256     = "0sx78pgx0qw8v7v2r32zzy3l161zipzq95iacda628girim7psnl";
+   "8.14.1".sha256     = "0xdqiabgm4lrm6d7lw544zd8xwb1cdcavsxvwwlqq6yid2rl2yli";
    "8.15.0".sha256     = "sha256:1ma76wfrpfsl72yh10w1ys2a0vi0mdc2jc79kdc8nrmxkhpw1nxx";
    "8.15.1".sha256     = "sha256:1dsa04jzkx5pw69pmxn0l55q4w88lg6fvz7clbga0bazzsfnsgd6";
-   "8.15.2".sha256     = "sha256:0gn8dz69scxnxaq6ycb3x34bjfk9wlp1y2xn8w69kg9fm4b6gkc7";
-   "8.16+rc1".sha256   = "sha256-dU+E0Mz7MVntbQIeG9I59ANBaHaXXSrjCRdoqZ5TO60=";
+   "8.15.2".sha256     = "sha256:0qibbvzrhsvs6w3zpkhyclndp29jnr6bs9i5skjlpp431jdjjfqd";
+   "8.16+rc1".sha256   = "sha256-hmZQ6rFIOZJwnAh23nKScJ3Nn+xqDRn5q2Tn82igpYE=";
```

Comment:
The utf8 file names were introduced in Coq 8.7 (https://github.com/coq/coq/pull/1041), so I'm expecting that all the `sha256` since 8.7.0 have to be recomputed. The patch-level versions that are not the latest are not exposed (they are only accessible by overriding the derivation) but they should still be fixed or removed (preferably the former).

Reply:
Sounds good. Updated rest of hashes. Only 8.13.0+ were broken possibly by 9ffd16b3850536094ca36bc31520bb15a6d5a9ef. A bit of investigation below:

Extracted broken ones with the following hack:

```diff
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -33270,6 +33270,7 @@ with pkgs;
     coqPackages_8_15 coq_8_15
     coqPackages_8_16 coq_8_16
     coqPackages      coq
+    mkCoq
   ;

   coq2html = callPackage ../tools/typesetting/coq2html { };
diff --git a/pkgs/top-level/coq-packages.nix b/pkgs/top-level/coq-packages.nix
index 91894bfaa9e..332a6d2bcb3 100644
--- a/pkgs/top-level/coq-packages.nix
+++ b/pkgs/top-level/coq-packages.nix
@@ -167,5 +167,5 @@ in rec {
   coqPackages_8_16 = mkCoqPackages coq_8_16;
   coqPackages = recurseIntoAttrs coqPackages_8_15;
   coq = coqPackages.coq;
-
+  inherit mkCoq;
 }
```

```
$ for v in 8.5pl1 8.5pl2 8.5pl3 8.6.0 8.6.0 8.6.1 8.7.0 8.7.1 8.7.2 8.8.0 8.8.1 8.8.2 8.9.0 8.9.1 8.10.0 8.10.1 8.10.2 8.11.0 8.11.1 8.11.2 8.12.0 8.12.1 8.12.2 8.13.0 8.13.1 8.13.2 8.14.0 8.14.1 8.15.0 8.15.1 8.15.2 8.16+rc1; do echo $v; nix build --impure --expr 'with import ./. {}; (mkCoq "'$v'").src'; nix build --impure --expr 'with import ./. {}; (mkCoq "'$v'").src' --rebuild -L; done
8.5pl1
```

Surprisingly (or not so surprisingly) breakage started only from `8.13.0` (UTF-8 files do present in older tarballs).

Unpatched `fetchzip` indeed already fails to unpack existing `8.12.0` (patches one accepts the checksum):

```
$ nix build --no-link -f. -f. coq_8_12.src --rebuild
error: hash mismatch in fixed-output derivation '/nix/store/0gvwnzj4dlx9bc78dj08ynh0d0l6gqcr-source.drv':
         specified: sha256-VnSw25bINfpJUkwulJZKtZUXVbYBYF0JhvimAapj+qE=
            got:    sha256-p+ZNPssu78SObNSPo/2N92qTvl72MvXbx00wUoggth4=
```

This means `unzip` (?) behavior changed somewhat recently (or packaging was done on non-glibc). Trying to bisect:

```
# commit adding 8.12.0 to nixpkgs:
$ git checkout b8dfca143c5b9e632530720ab472bd54d9d21137
$ nix build --no-link -f. -f. coq_8_12.src --rebuild
# OK!

$ git bisect start origin/master b8dfca143c5b9e632530720ab472bd54d9d21137
$ git bisect run nix build --no-link -f. -f. coq_8_12.src --rebuild

commit 9ffd16b3850536094ca36bc31520bb15a6d5a9ef
Date:   Fri Aug 28 23:05:46 2020 +0200

    coqPackages: refactor
```

Looks like 9ffd16b3850536094ca36bc31520bb15a6d5a9ef did change from `fetchFromGitHub` (`.tar.gz`) to `fetchzip` (`.zip`) and introduced the discrepancy.